js/ag_reference.js

```diff
@@ -371,7 +484,30 @@ AttributeGroupReference.prototype = {
         });
 
         return defer.promise;
-    }
+    },
+
+    /**
+     * Find a column in list of key and aggregate columns.
+     * @param  {string} name the column name
+     * @return {ERMrest.AttributeGroupColumn}
+     */
+    getColumnByName: function (name) {
+
+        var findCol = function (list) {
+            for (i = 0; i < list.length; i++) {
+                if (list[i].name === name) {
+                    return list[i];
+                }
+            }
+            return false;
+        };
+
+        var c = findCol(this._allColumns);
```

Comment:
if `_allColumns` is a `Columns` object like we have defined in `core.js`, this function is redundant because of `Columns.get()`.

Reply:
It's not. It's an array of `AttributeGroupColumn` object which is different from `Column`. Besides I didn't follow the pattern of creating objects for each array container like we have in `core.js` (So the `_allColumns` is actually an array and not an object that contains an array). It might look better, but it's just redundant and reduces the flexibility.